corehq/apps/app_manager/app_translations/validator.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import absolute_import
+
+from memoized import memoized
+from corehq.apps.app_manager.app_translations import (
+    expected_bulk_app_sheet_headers,
+    expected_bulk_app_sheet_rows,
+    get_unicode_dicts,
+)
+from corehq.apps.app_manager.app_translations.const import MODULES_AND_FORMS_SHEET_NAME
+
+COLUMNS_TO_COMPARE = {
+    'module_and_form': ['Type', 'sheet_name'],
+    'module': ['case_property', 'list_or_detail'],
+    'form': ['label'],
+}
+ROW_COUNT_MISMATCH_MESSAGE = "We found {} rows than expected. Validating what was uploaded."
+
+
+class UploadedTranslationsValidator(object):
```

Comment:
Are these new validations? Are they errors that currently break bulk translations but we don't surface them to users?

Reply:
> Are they errors that currently break bulk translations

I think some of them would still get fetched for example [here](https://github.com/dimagi/commcare-hq/blob/1b6bdda8bd5c4dad18f8abc0ce99e3ad010991f2/corehq/apps/app_manager/app_translations/app_translations.py#L914) but then the rest of the update would happen in the current process. Down the road we can probably combine these two implementations in a way.. but this one is just to automate the manual process one would follow to compare two excel sheets(expected vs to be uploaded) and share possible discrepancies.

Reply:
> Down the road we can probably combine these two implementations in a way

This is the kind of thing that doesn't tend to happen for a while, if ever, and gets harder to do the more time passes. It seems unfortunate that you've spent the time on this feature but that it'll only benefit ICDS, when bulk translation errors are a known pain point for other projects.